Tests/BSONTests/BSONCorpusTests.swift

```diff
@@ -129,97 +156,101 @@ final class BSONCorpusTests: BSONTestCase {
 
-                    if testFile.description == "Decimal128" {
-                        // TODO: This should be tested by EXTJSON
-                        struct Decimal128CanonicalExtJSON: Codable {
-                            struct Value: Codable {
-                                enum CodingKeys: String, CodingKey {
-                                    case numberDecimal = "$numberDecimal"
-                                }
+                    // native_to_canonical_extended_json( bson_to_native(cB) ) = cEJ
+                    let canonicalEncoder = ExtendedJSONEncoder()
+                    canonicalEncoder.mode = .canonical
+                    expect(try canonicalEncoder.encode(docFromCB))
+                        .to(cleanEqual(test.canonicalExtJSON), description: test.description)
 
-                                var numberDecimal: String
-                            }
+                    // native_to_relaxed_extended_json( bson_to_native(cB) ) = rEJ (if rEJ exists)
+                    let relaxedEncoder = ExtendedJSONEncoder() // default mode is .relaxed
+                    if let rEJ = test.relaxedExtJSON {
+                        expect(try relaxedEncoder.encode(docFromCB))
+                            .to(cleanEqual(rEJ), description: test.description)
+                    }
 
-                            var d: Value
+                    // for cEJ input:
+                    // native_to_canonical_extended_json( json_to_native(cEJ) ) = cEJ
+                    let decoder = ExtendedJSONDecoder()
+                    expect(try canonicalEncoder.encode(try decoder.decode(BSON.self, from: cEJData)))
+                        .to(cleanEqual(test.canonicalExtJSON), description: test.description)
+
+                    // native_to_bson( json_to_native(cEJ) ) = cB (unless lossy)
+                    if !lossy {
+                        try expect(try decoder.decode(BSONDocument.self, from: cEJData))
+                            .to(sortedEqual(BSONDocument(fromBSON: cBData)), description: test.description)
+                    }
+
+                    // for dB input (if it exists): (change to language native part)
+                    if let dB = test.degenerateBSON {
+                        guard let dBData = Data(hexString: dB) else {
+                            XCTFail("Unable to interpret degenerate_bson as Data")
+                            return
                         }
-                        let extjson = test.canonicalExtJSON.data(using: .ascii)!
-                        let jsonResult = try JSONDecoder().decode(Decimal128CanonicalExtJSON.self, from: extjson)
-                        let decimal128CorpusString = jsonResult.d.numberDecimal
 
-                        let decimal128FromString = try BSONDecimal128(decimal128CorpusString)
-                        let decimal128FromBinary = docFromCB.d!.decimal128Value!
+                        // bson_to_canonical_extended_json(dB) = cEJ
```

Comment:
This section and the next one for `bson_to_relaxed_extended_json` can be deleted.